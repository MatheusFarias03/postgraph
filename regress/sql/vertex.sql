/*
 * PostGraph
 * Copyright (C) 2023 by PostGraph
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

LOAD 'age';
SET search_path TO ag_catalog;

--
-- Vertex
--
--Basic Vertex Creation
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map());
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));

--Null properties
SELECT build_vertex('1'::graphid, $$label_name$$, NULL);
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_list());


--
-- id function
--
SELECT id(build_vertex('1'::graphid, $$label$$, agtype_build_map()));
SELECT id(NULL);

--
-- label function
--
SELECT label(build_vertex('1'::graphid, $$label$$, agtype_build_map()));
SELECT label(NULL);

--
-- -> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map())->'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->'idd';

--
-- ->> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map())->>'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->>'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->>'idd';

--
-- @> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) @> agtype_build_map();
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map();
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map('id', 2);
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map('id', 1);

--
-- <@ operator
--
SELECT agtype_build_map() <@ build_vertex('1'::graphid, $$label$$, agtype_build_map());
SELECT agtype_build_map() <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
SELECT agtype_build_map('id', 2) <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
SELECT agtype_build_map('id', 1) <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));

--
-- ? operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ? 'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ? 'id';
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ? 'idd';

--
-- ?| operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ?| array['id'];
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?| array['id'];
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?| array['idd'];

--
-- ?& operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ?& array['id'];
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?& array['id'];
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?& array['idd'];


