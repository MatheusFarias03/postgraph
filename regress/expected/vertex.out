/*
 * PostGraph
 * Copyright (C) 2023 by PostGraph
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
LOAD 'age';
SET search_path TO ag_catalog;
--
-- Vertex
--
--Basic Vertex Creation
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map());
                 build_vertex                  
-----------------------------------------------
 {"id": 1, "label": "label", "properties": {}}
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
                     build_vertex                     
------------------------------------------------------
 {"id": 1, "label": "label", "properties": {"id": 2}}
(1 row)

--Null properties
SELECT build_vertex('1'::graphid, $$label_name$$, NULL);
 build_vertex 
--------------
 
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_list());
 build_vertex 
--------------
 
(1 row)

--
-- id function
--
SELECT id(build_vertex('1'::graphid, $$label$$, agtype_build_map()));
 id 
----
 1
(1 row)

SELECT id(NULL);
 id 
----
 
(1 row)

--
-- label function
--
SELECT label(build_vertex('1'::graphid, $$label$$, agtype_build_map()));
  label  
---------
 "label"
(1 row)

SELECT label(NULL);
 label 
-------
 
(1 row)

--
-- -> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map())->'id';
 ?column? 
----------
 
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->'id';
 ?column? 
----------
 2
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->'idd';
 ?column? 
----------
 
(1 row)

--
-- ->> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map())->>'id';
 ?column? 
----------
 
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->>'id';
 ?column? 
----------
 2
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2))->>'idd';
 ?column? 
----------
 
(1 row)

--
-- @> operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) @> agtype_build_map();
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map();
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map('id', 2);
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) @> agtype_build_map('id', 1);
 ?column? 
----------
 f
(1 row)

--
-- <@ operator
--
SELECT agtype_build_map() <@ build_vertex('1'::graphid, $$label$$, agtype_build_map());
 ?column? 
----------
 t
(1 row)

SELECT agtype_build_map() <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
 ?column? 
----------
 f
(1 row)

SELECT agtype_build_map('id', 2) <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
 ?column? 
----------
 t
(1 row)

SELECT agtype_build_map('id', 1) <@ build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2));
 ?column? 
----------
 f
(1 row)

--
-- ? operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ? 'id';
 ?column? 
----------
 f
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ? 'id';
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ? 'idd';
 ?column? 
----------
 f
(1 row)

--
-- ?| operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ?| array['id'];
 ?column? 
----------
 f
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?| array['id'];
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?| array['idd'];
 ?column? 
----------
 f
(1 row)

--
-- ?& operator
--
SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map()) ?& array['id'];
 ?column? 
----------
 f
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?& array['id'];
 ?column? 
----------
 t
(1 row)

SELECT build_vertex('1'::graphid, $$label$$, agtype_build_map('id', 2)) ?& array['idd'];
 ?column? 
----------
 f
(1 row)

